### ISSUE TITLE
0.11.x MatchMake filter should support more query operations

### ISSUE THREAD

**Issue by GDdark** [colyseus/colyseus#259]
client.joinOrCreate("battle", { mode: "duo" }).then(room => {/* ... */});

It looks like I can only match special rooms which's properties is equal to conditions. But that's not enough. for example, i want rooms which's clients > 3.

So I have to install mongodb to extend this
But I think it's too heavy to install  mongodb just for these functions.... 
Supported in local query drive is best!

**Comment by RonnyHou** [comment]
agree. @endel should there be a customized filtering procedure where I can achieve options condition like: not equal, contains, not contain, greater, less etc. ?

**Comment by endel** [comment]
Closing as not going to implement this for now. Only a major restructuring of matchmaking would allow this.
